message/infrastructure/nats/subscriber.go

```diff
@@ -391,3 +394,10 @@ func (s *StreamingSubscriber) Close() error {
 
 	return result
 }
+
+func (s *StreamingSubscriber) isClosed() bool {
+	s.subsLock.RLock()
+	defer s.subsLock.RUnlock()
+
+	return s.closed
```

Comment:
can you explain why you added the rmutex here?

Reply:
`s.closed` can be concurrently read in `processMessage` and written in `Close()` during this, and we have a race condition